takari-builder/src/main/java/io/takari/builder/internal/cache/ScopedProjectDependencyCache.java

```diff
@@ -0,0 +1,91 @@
+package io.takari.builder.internal.cache;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Supplier;
+
+import javax.inject.Named;
+
+import org.apache.maven.SessionScoped;
+import org.apache.maven.artifact.Artifact;
+
+import io.takari.builder.ResolutionScope;
+
+
+@Named
+@SessionScoped
+public class ScopedProjectDependencyCache {
```

Comment:
I would java doc this.   Seems a bit important, no?  This seems a represtantion of the project's deps. Not necessarily the deps you want exposed to the builder -- but I'm sure I'll see that class in a bit.